### ISSUE TITLE
Virtual keyboard overlaps map view when choosing location to send

### ISSUE THREAD

**Issue by e-caste** [nextcloud/spreed-ios#792]
## Steps to reproduce
1. Open chat with someone
2. Open keyboard by tapping on text field
3. Choose attachments button
4. Choose location
5. Map opens partially covered by keyboard (most importantly, the buttons to choose the location to send are covered)
 
### Expected behaviour
The map should be visible.

### Actual behaviour
The map and the buttons are not visible because they are covered by the virtual keyboard and there is no method to dismiss it from the map view. 

#### Notes:
- in the right screenshot I have **not** tapped on the search bar
- this bug seems to only affect the location attachment, I have tried with all the other types but they work fine even with an open keyboard in the chat view

| If I open the location screen without tapping on the text field beforehand (keyboard closed)  | If I open the location screen having tapped on the text field beforehand (keyboard open) |
|--------------|-----------|
| ![IMG_6033](https://user-images.githubusercontent.com/48513706/167591252-45b414ec-d133-4d6d-bf3b-1792d5ac2242.PNG) | ![IMG_6034](https://user-images.githubusercontent.com/48513706/167591302-8eb4feaf-17cf-4479-ab26-666d76bd7cf3.PNG) |

## Device information

**Device:** iPhone X

**iOS version:** 15.4.1

**Talk version:** 14.0.0

## Server information

**Nextcloud version:** 24.0.0

**Talk version:** 14.0.1

**Custom Signaling server configured:** no

**Custom TURN server configured:** yes

**Custom STUN server configured:** yes

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

**Comment by SystemKeeper** [comment]
Hey @e-caste 
Can you please re-check if this still occurs on the latest version? I tried to reproduce this with talk-ios 14.0.1 and 14.0.2 on a phone (11 pro) and on simulator and can't reproduce the issue you're seeing. Although I did see another problem with location sharing (#810)

**Comment by e-caste** [comment]
@SystemKeeper yep, can still reproduce the issue with NCT iOS 14.0.2:
1. Open a chat
2. Open the keyboard without typing anything
3. Choose attachments -> location
4. Allow once for location
5. The keyboard overlaps the map

**Comment by SystemKeeper** [comment]
Hm, sorry, can't reproduce this on a real phone, nor on simulator:

https://user-images.githubusercontent.com/1580193/170594183-c48db610-a662-4d2d-9099-9f13e348df7f.mp4

We could work around this by always dismissing the keyboard when pressing the attachment button (as noted in #810), but I'm not sure why this happens on your device right now.

**Comment by e-caste** [comment]
@SystemKeeper have you tried the same steps after setting the location settings for Nextcloud Talk to this? I think this may be the culprit.
![image](https://user-images.githubusercontent.com/48513706/170641580-67d5c3f3-49df-47b3-b20b-22ba80c607a2.png)

**Comment by SystemKeeper** [comment]
Pretty sure I did this yesterday, but retrying today I see the problem. The problem arises because the app looses and then regains focus. 
You can also trigger this problem switching back and forth to the app, while the location sharing (or photo sharing) is opened. Then the keyboard will be shown after coming back.

**Comment by e-caste** [comment]
@SystemKeeper yes I can reproduce the "keyboard overlapping any share view" after going to the home screen and back to the app when I leave the keyboard up. I think either this issue should be extended to encapsulate this problem as well or a separate issue is needed.

**Comment by SystemKeeper** [comment]
I think it's fine in this issue, as the root cause is the same. I'll take a look how we can fix this without hiding the keyboard everytime the app moves to the background.

**Comment by SystemKeeper** [comment]
Hey @e-caste 
Do you have upgraded to iOS 16 already? If yes, can you please retest this as I'm not able to reproduce this anymore.

**Comment by e-caste** [comment]
@SystemKeeper yes, with iPhone X on iOS 16.0.2 and Nextcloud Talk iOS 14.0.3 I can still reproduce the issue following the steps of my previous comment.
